components/p3a/nitro_utils/cose.h

```diff
@@ -0,0 +1,44 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_P3A_NITRO_UTILS_COSE_H_
+#define BRAVE_COMPONENTS_P3A_NITRO_UTILS_COSE_H_
+
+#include <vector>
+
+#include "components/cbor/values.h"
+#include "net/cert/pki/verify_certificate_chain.h"
+
+namespace nitro_utils {
+
+class CoseSign1 : net::VerifyCertificateChainDelegate {
+ public:
+  CoseSign1();
+  ~CoseSign1() override;
+
+  bool DecodeFromBytes(const std::vector<uint8_t>& data);
+
+  bool Verify(const net::ParsedCertificateList& cert_chain);
+
+  cbor::Value protected_headers;
```

Comment:
pls make it CC-compatible, i.e. move member fields to the private section, and `= delete` copy/assignment